e3nn/util/codegen/_mixin.py

```diff
@@ -31,10 +33,17 @@ def _codegen_register(
             self.__codegen__ = []
         self.__codegen__.extend(funcs.keys())
 
+        opt_defaults = e3nn.get_optimization_defaults()
```

Comment:
see discussing in Slack; TensorProduct should really take a `optimization_options` dict and should propagate it through to CodeGenMixin so that it is possible to override this on an object-by-object basis

low importance tho

Reply:
Also agree but I think we can do that in a separate PR

Reply:
👍